Gems/EMotionFX/Code/Source/Editor/Plugins/Ragdoll/PhysicsSetupViewportUiCluster.cpp

```diff
@@ -71,6 +80,10 @@ namespace EMotionFX
             m_buttonIds.resize(static_cast<size_t>(SubMode::NumModes));
             m_buttonIds[static_cast<size_t>(SubMode::ColliderTranslation)] = RegisterClusterButton(viewportId, m_clusterId, "Move");
             m_buttonIds[static_cast<size_t>(SubMode::ColliderRotation)] = RegisterClusterButton(viewportId, m_clusterId, "Rotate");
+            if (m_hasCapsuleCollider)
+            {
+                m_buttonIds[static_cast<size_t>(SubMode::ColliderDimensions)] = RegisterClusterButton(viewportId, m_clusterId, "Scale");
+            }
```

Comment:
question: Let's say the first call to this function there is a capsule collider (`m_hasCapsuleCollider` is true) and the `m_buttonIds[ColliderDimension]` gets assigned a register cluster button. Then in a second call there is no capsule collider (`m_hasCapsuleCollider` is false), all the other elements of `m_buttonIds` are overwritten with new `RegisterClusterButton` calls, but `m_buttonIds[ColliderDimension]` will keep the same one from the first call because it won't enter inside this `if`. Is that an issue?

Reply:
Good spot and that isn't ideal, but I don't think it can be a problem because the only way to activate one of the modes at the moment is to click on a button, and the button won't exist in that case. That logic is only temporary until the other geometries (box and sphere) are supported for dimension manipulators.